spec/abci/abci++_app_requirements.md

```diff
@@ -188,6 +188,31 @@ Likewise, `ExtendVote` can also be non-deterministic:
 * *w<sup>r</sup><sub>p</sub> = w<sup>r</sup><sub>q</sub> &#8655;
   e<sup>r</sup><sub>p</sub> = e<sup>r</sup><sub>q</sub>*
 
+Unlike all requirements described above,
+which affect the ABCI methods of the [Consensus Connection](#consensus-connection),
+the following requirement is on the [Mempool Connection](#mempool-connection).
+Let *CheckTxCode<sub>tx,p,t</sub>* denote the error code returned by *p*'s Application, via `ReponseCheckTx`,
+to CometBFT's `RequestCheckTx` call occurring at time *t* and having transaction *tx* as parameter.
+Let predicate *OK(CheckTxCode)* denote whether *CheckTxCode* is `SUCCESS`.
+
+
+* Requirement 13 [`CheckTx`, non-oscillation]: For any correct process *p*
+  and any transaction *tx*,
+  (a) either *OK(CheckTxCode<sub>tx,p,t<sub>0</sub></sub>) = OK(CheckTxCode<sub>tx,p,t<sub>1</sub></sub>)*
+  for any two times *t<sub>0</sub>* and *t<sub>1</sub>* in *p*'s execution, or
+  (b) there exists a time *t<sub>stable</sub>* in *p*'s execution
+  such that *&#172; OK(CheckTxCode<sub>tx,p,t</sub>)* for all *t > t<sub>stable</sub>*.
+
+Requirement 13 has two main conditions. The implementation of `CheckTx` must fullfill at least one of them.
+Condition (a) mandates that `CheckTx` on a transaction never changes its result,
+in other words, that the checks implemented are stateless (e.g., well-formedness, signature verification).
+Condition (b) states that, if the transaction stays in the mempool long enough, if will eventually fail
```

Comment:
> if the transaction stays in the mempool long enough

Does it have to stay in the mempool or could it be added and removed repeatedly before the stabilization?

Reply:
That's what I understand we're trying to avoid: some nodes saying it's valid, gossiping it, then saying it is invalid... and the TX circulating around the p2p network but always disappearing them moment it is to be proposed.
Check the latest version, where I try to clarify that.

Reply:
If the transaction is removed and added again, it will need to pass through CheckTx, with the possibility to be rejected the second time. If it stays "long enough" in the mempool, without being rechecked, most probably it will eventually become invalid, while in the mempool.